### DISCUSSION TITLE
[Alist衍生]自建pdf.js的问题

### DISCUSSION THREAD

**Discussion by MrZhongzq** [AlistGo/alist#5432] [General]
如题, 由于访问alist官方的pdf服务困难,我在自己服务器上搭建了一个pdf.js供alist使用.
现在pdf.js显示不出文件,文件本身读取代码200,可以访问. 网站: https://pdf.zzqhmbb.eu.org/web/viewer.html
<img width="608" alt="image" src="https://github.com/alist-org/alist/assets/108169409/3f9a35cd-dd73-47a0-b7d8-3d625400db3b">
<img width="1423" alt="image" src="https://github.com/alist-org/alist/assets/108169409/8c749dc6-1b7a-4c98-a3fe-217c90efff5b">
使用alist配合其他人的pdf.js可以正常加载,排除了alist问题.
alist 版本:v3.28  pdf.js版本: v3.11.174
pdf.js即使手动加载本地文件也不可以.

**Comment by gmugu** [comment]
应该是跨域拦截了，如果你的pdf.js 是从alist-org.github.io/pdf.js clone 的，可以把代码中“alist-org.github.io”替换成你自己的域名

**Reply by MrZhongzq** [reply]
我把跨域改了,现在手动启动pdf.js上传文件已经正常,但是从alist启动padjs预览还是白屏. 我使用了pdf.js作者的源站就可以从alist调用

**Reply by gmugu** [reply]
![image](https://github.com/alist-org/alist/assets/8317478/c30d5872-a328-4c7f-82f7-668713859ffb)
资源加载不全，你的部署有问题？

**Reply by 396647599** [reply]
有没有office的呢

**Comment by MirageTurtle** [comment]
由于我没有https，看了一些issue，看上去自建一个pdf.js可以使用，但我不太会设置如何使用自建的pdf.js，想请教一下。

我看了[这里](https://alist.nn.ci/zh/config/preview.html)的iframe，应该是在这里设置pdf类型的预览设置吧？但我是个新手，没太看懂这个配置应该在哪里修改，也有可能我忽略了一些文档，还烦请指出。

**Comment by MrZhongzq** [comment]
我用的是作者的预构建的版本,不是clone的,直接就是没有src这个文件夹的. 我下载下来只有web和build.

MuGu ***@***.***> 于2023年10月31日周二 08:46写道：

**Reply by gmugu** [reply]
git clone -b main https://github.com/alist-org/pdf.js.git
用这个，然后把"alist-org.github.io"替换成你自己的域名，测试过是可以的。

**Reply by hankqqu** [reply]
> git clone -b main https://github.com/alist-org/pdf.js.git
> 
> 用这个，然后把"alist-org.github.io"替换成你自己的域名，测试过是可以的。

**Comment by MrZhongzq** [comment]
用alist版的pdf js可以使用了


MuGu ***@***.***> 于2023年10月31日周二 14:11写道：

**Reply by sbbbsbbb** [reply]
他这里的改alist-org.github.io替换成你自己的域名 是在iframe 那边改吗，还是改alist版的pdf js里面的？
